chirp/drivers/icm710.py

```diff
@@ -0,0 +1,178 @@
+        rf.valid_bands = [(1000000, 29999999)]
+
+        rf.valid_duplexes = DUPLEX_MODES
+        rf.can_odd_split = True
+        rf.valid_modes = MODULATION_MODES
+        rf.valid_power_levels = POWER_LEVELS
+        rf.valid_tones = []
+        rf.valid_tmodes = []
+        rf.valid_skips = []
+        return rf
+
+    # Convert the raw byte array into a memory object structure
+    def process_mmap(self):
+        self._memobj = bitwise.parse(MEM_FORMAT, self._mmap)
+
+    # Return a raw representation of the memory object, which
+    # is very helpful for development
+    def get_raw_memory(self, number):
+        return repr(self._memobj.memory[number])
+
+    def sync_out(self):
+        self.pipe.baudrate = 4800
+        icf.clone_to_radio(self)
+
+    # Extract a high-level memory object from the
+    # low-level memory map
+    # This is called to populate a memory in the UI
+    def get_memory(self, number):
+
+        # this just gets the 160 common
+        # the 72 ITU channels are probably working
+        # there may be some strange behavior with
+        # the SITOR FSK "hidden" channels
+        # best be careful about trying to set a fixed
+        # simplex channel with a duplex frequency
+        # TODO test, test, test, all the combinations.
+
+        # Get a low-level memory object mapped to the image
+        _mem = self._memobj.memory[int(number)]
+
+        # Create a high-level memory object to return to the UI
+        mem = chirp_common.Memory()
+
+        mem.number = number
+        mem.comment = ''
+
+        # Convert your low-level frequency to Hertz.
+        mem.freq = int(_mem.rx_freq)
+        mem.name = str(_mem.name).rstrip()  # Set the alpha tag
+
+        mem.mode = MODULATION_MODES[0]
+        if (_mem.mode > -1) and (_mem.mode < 7):
+            mem.mode = MODULATION_MODES[int(_mem.mode)]
+
+        # if the frequency is above 30MHz
+        # (166666665 or FF,FF,FF,FF), it's simplex
+        mem.offset = int(_mem.tx_freq)
+        if int(_mem.tx_freq) < 30000000:
+            mem.duplex = 'split'
+            mem.comment = 'TX: ' + \
+                str((int(_mem.tx_freq)/1000000)) + \
+                ' MHz'
+        else:
+            mem.duplex = 'off'
+            mem.comment = ''
+            # _mem.tx_freq.set_raw('\xFF\xFF\xFF\xFF')
+            _mem.tx_freq.set_raw(bytes(b'\xFF')*4)
+
+        # We'll consider any blank (i.e. 0 MHz frequency) to be empty
+        if (mem.freq < 1000000) or (mem.freq > 29999999):
+            mem.empty = True
+
+        return mem
+
+    # Store details about a high-level memory to the memory map
+    # This is called when a user edits a memory in the UI
+    def set_memory(self, mem):
+        # Get a low-level memory object mapped to the image
+        _mem = self._memobj.memory[int(mem.number)]
+
+        _mem.name = mem.name.ljust(7)[:7]
+
+        if mem.empty == 1:
+            _mem.rx_freq.set_raw(bytes(b'\xFF')*4)
```

Comment:
a `b'foo'` is already a `bytes`. Doing this just wastes cpu cycles. Just `b'\xFF' * 4` is good. Also, please use spaces around operators.

But otherwise, thanks for converting these to `bytes` 👍